modules/monitoring/controllers/pnp.php

```diff
@@ -49,9 +49,32 @@ public function pnp_default()
 		if ($pnp_path != '') {
 			$source = intval($this->input->post('source', false));
 			$view = intval($this->input->post('view', false));
+			$settings = json_encode(array("source"=>$source, "view"=>$view));
 
-			Ninja_setting_Model::save_page_setting('source', $pnp_path.'/image?'.$param, $source);
-			Ninja_setting_Model::save_page_setting('view', $pnp_path.'/image?'.$param, $view);
+			Ninja_setting_Model::save_page_setting('default_graph', $pnp_path.'/image?'.$param, $settings);
 		}
 	}
+
+	/**
+	 * Get graph setting for a specific host/service
+	 */
+	public function get_pnp_default()
+	{
+
+		/* Ajax calls shouldn't be rendered. This doesn't, because some unknown
+		 * magic doesn't render templates in ajax requests, but for debugging
+		 */
+		$this->auto_render = false;
+
+		$page = $this->input->post('page', false);
+
+		$pnp_path = Kohana::config('config.pnp4nagios_path');
+
+		if ($pnp_path != '') {
+			$settings = Ninja_setting_Model::fetch_page_setting('default_graph', $page);
+		}
+		// Returns false on no entry found
+		echo $settings->setting;
```

Comment:
Put this inside the `if` block above. Echo some default here if `$pnp_path` isn't set?

Reply:
done